pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -619,6 +623,9 @@ func ValidateManagedNodeGroup(ng *ManagedNodeGroup, index int) error {
 		if !IsAMI(ng.AMI) {
 			return errors.Errorf("invalid AMI %q (%s.%s)", ng.AMI, path, "ami")
 		}
+		if ng.AMIFamily != NodeImageFamilyAmazonLinux2 {
+			return errors.Errorf("cannot set amiFamily to %s when using a custom AMI", ng.AMIFamily)
+		}
```

Comment:
does this need to include bottlerocket? slightly confused

Reply:
The alternative would be to include all the unsupported AMI families. I'll try to make this more explicit in the PR for Ubuntu support.

Reply:
```
if ng.AMIFamily != NodeImageFamilyAmazonLinux2 {			
  return errors.Errorf("cannot set amiFamily to %s when using a custom AMI", ng.AMIFamily)		
}
```

wait but doesn't this mean that if you provide bottlerocket, it will fail?

Reply:
Yes, that's the intent. I meant the alternative would to be do a `if ng.AMIFamily != unsupported AMI families`.